spring-integration-core/src/main/java/org/springframework/integration/config/xml/AbstractConsumerEndpointParser.java

```diff
@@ -168,4 +175,19 @@ protected final AbstractBeanDefinition parseInternal(Element element, ParserCont
 		return null;
 	}
 
+	/**
+	 * Override to allow 'reply-channel' within a chain, for components where it
+	 * makes sense (e.g. enricher). Default is false for outbound gateways, else true.
+	 * @return true to allow a reply channel attribute within a chain.
+	 */
+	protected boolean replyChannelInChainAllowed(Element element) {
+		String localName = element.getLocalName();
+		if (localName == null) {
```

Comment:
It is `null` here for the 
```
	try {
				gatherer = documentBuilderFactory.newDocumentBuilder().newDocument().createElement("aggregator");
			}
```
See `ScatterGatherParser`.
Even if we understand why and can fix it there, it doesn't matter that we won't meet similar issue for some other case in the future.